### PR TITLE
View basic info and Secrity Model

### PR DESCRIPTION
**Pull request by OlehHatsko**
dev
## JIRA

* [Main JIRA ticket](https://jira.softserve.academy/secure/RapidBoard.jspa?rapidView=id)

## Code reviewers

- [ ] @github_username

### Second Level Review

- [ ] @github_username

## Summary of issue

ToDo

## Summary of change

ToDo

## Testing approach

ToDo

## CHECK LIST
- [ ]  СI passed
- [ ]  Сode coverage >=95%
- [ ]  PR is reviewed manually again (to make sure you have 100% ready code)
- [ ]  All reviewers agreed to merge the PR
- [ ]  I've checked new feature as logged in and logged out user if needed
- [ ]  PR meets all conventions

### CODE REVIEW

File: EPlast/EPlast.BLL/DTO/GoverningBody/GoverningBodyProfileDTO.cs [hunk 1]
```diff
@@ -0,0 +1,10 @@
+﻿using System.Collections.Generic;
+
+namespace EPlast.BLL.DTO.GoverningBody
+{
+    public class GoverningBodyProfileDTO
+    {
+        public GoverningBodyDTO GoverningBody { get; set; }
+        public List<GoverningBodyDocumentsDTO> Documents { get; set; }
```
Comment (voroniak) [inline EPlast/EPlast.BLL/DTO/GoverningBody/GoverningBodyProfileDTO.cs:8]: It's better to use ienumerable

File: EPlast/EPlast.BLL/Interfaces/GoverningBodies/IGoverningBodiesService.cs [hunk 2]
```diff
@@ -11,5 +13,13 @@ public interface IGoverningBodiesService
...
 
         Task<string> GetLogoBase64(string logoName);
+
+        Task<GoverningBodyProfileDTO> GetProfileById(int id, User user);
+
+        Task<int> RemoveAsync(int governingBodyId);
+
+        Task<int> EditAsync(GoverningBodyDTO governingBody);
+
+        Task<Dictionary<string, bool>> GetUserAccess(string userId);
```
Comment (voroniak) [inline EPlast/EPlast.BLL/Interfaces/GoverningBodies/IGoverningBodiesService.cs:23]: Async

File: EPlast/EPlast.BLL/Interfaces/SecurityModel/ISecurityModel.cs [hunk 3]
```diff
@@ -0,0 +1,11 @@
+﻿using System.Collections.Generic;
+using System.Threading.Tasks;
+
+namespace EPlast.BLL.Interfaces
+{
+    public interface ISecurityModel
+    {
+        Task<Dictionary<string, bool>> GetUserAccess(string userId, IEnumerable<string> userRoles= null);
```
Comment (voroniak) [inline EPlast/EPlast.BLL/Interfaces/SecurityModel/ISecurityModel.cs:8]: Async

File: EPlast/EPlast.BLL/Services/GoverningBodies/GoverningBodiesService.cs [hunk 4]
```diff
@@ -80,9 +97,42 @@ private async Task UploadPhotoAsync(GoverningBodyDTO governingBody)
 
         public async Task<string> GetLogoBase64(string logoName)
         {
-            var logoBase64 = await _governingBodyBlobStorage.GetBlobBase64Async(logoName);
+            return await _governingBodyBlobStorage.GetBlobBase64Async(logoName);
+        }
+
+        public async Task<GoverningBodyProfileDTO> GetProfileById(int id, User user)
```
Comment (voroniak) [inline EPlast/EPlast.BLL/Services/GoverningBodies/GoverningBodiesService.cs:103]: Async

File: EPlast/EPlast.BLL/Services/GoverningBodies/GoverningBodiesService.cs [hunk 5]
```diff
@@ -80,9 +97,42 @@ private async Task UploadPhotoAsync(GoverningBodyDTO governingBody)
...
+            GoverningBodyProfileDTO gbProfile = new GoverningBodyProfileDTO
+            {
+                GoverningBody = await GetProfileById(id),
+            };
+
+            return gbProfile;
+        }
 
-            return logoBase64;
+        public async Task<GoverningBodyDTO> GetProfileById(int id)
```
Comment (voroniak) [inline EPlast/EPlast.BLL/Services/GoverningBodies/GoverningBodiesService.cs:113]: Async

File: EPlast/EPlast.BLL/Services/RegionBoard/RegionsBoardService.cs [hunk 6]
```diff
@@ -0,0 +1,26 @@
...
+using System.Collections.Generic;
+using System.Threading.Tasks;
+using EPlast.BLL.Interfaces;
+
+namespace EPlast.BLL.Services
+{
+    public class RegionsBoardService : IRegionsBoardService
+    {
+        private readonly ISecurityModel _securityModel;
+        private const string SecuritySettingsFile = "RegionsBoardAccessSettings.json";
```
Comment (voroniak) [inline EPlast/EPlast.BLL/Services/RegionBoard/RegionsBoardService.cs:12]: Is it correct naming?

File: EPlast/EPlast.WebApi/Controllers/GoverningBodiesController.cs [hunk 7]
```diff
@@ -45,13 +51,65 @@ public async Task<IActionResult> Create(GoverningBodyDTO governingBodyDTO)
...
+        {
+            if (!ModelState.IsValid)
+            {
+                return BadRequest(ModelState);
+            }
+
+            await _governingBodiesService.EditAsync(governingBody);
+            _logger.LogInformation($"Governing body {{{governingBody.GoverningBodyName}}} was edited.");
+
+            return Ok();
```
Comment (voroniak) [inline EPlast/EPlast.WebApi/Controllers/GoverningBodiesController.cs:66]: Why did you return empty ok result?

File: EPlast/EPlast.WebApi/Controllers/GoverningBodiesController.cs [hunk 8]
```diff
@@ -45,13 +51,65 @@ public async Task<IActionResult> Create(GoverningBodyDTO governingBodyDTO)
...
+        }
+
+        [HttpDelete("RemoveGoverningBody/{governingBodyId}")]
+        [Authorize(AuthenticationSchemes = "Bearer")]
+        public async Task<IActionResult> Remove(int governingBodyId)
+        {
+            await _governingBodiesService.RemoveAsync(governingBodyId);
+            _logger.LogInformation($"GoverningBody with id {{{governingBodyId}}} was deleted.");
+
+            return Ok();
```
Comment (voroniak) [inline EPlast/EPlast.WebApi/Controllers/GoverningBodiesController.cs:103]: I'm not sure is it ok to return 200 for delete

File: EPlast/EPlast.WebApi/Controllers/RegionsBoardController.cs [hunk 9]
```diff
@@ -0,0 +1,28 @@
...
+            _regionsBoardService = service;
+        }
+
+        [HttpGet("GetUserAccesses/{userId}")]
+        [Authorize(AuthenticationSchemes = "Bearer")]
+        public async Task<IActionResult> GetUserAccess(string userId)
+        {
+            var accesses = await _regionsBoardService.GetUserAccessAsync(userId);
+
+            return Ok(accesses);
```
Comment (voroniak) [inline EPlast/EPlast.WebApi/Controllers/RegionsBoardController.cs:25]: You may write it as 1 line

**Review by voroniak — CHANGES_REQUESTED** [review]

File: EPlast/EPlast.BLL/Interfaces/GoverningBodies/IGoverningBodiesService.cs (hunk 2 above)
Comment (OlehHatsko) [reply EPlast/EPlast.BLL/Interfaces/GoverningBodies/IGoverningBodiesService.cs:23]: Do you mean to add the `async` keyword to the method declaration? if the answer is yes, then the async keyword can only be used in methods that  have a body

File: EPlast/EPlast.BLL/Services/RegionBoard/RegionsBoardService.cs (hunk 6 above)
Comment (OlehHatsko) [reply EPlast/EPlast.BLL/Services/RegionBoard/RegionsBoardService.cs:12]: It`s constant, so I used PascalCase. Or did you mean anything else?

File: EPlast/EPlast.WebApi/Controllers/GoverningBodiesController.cs (hunk 7 above)
Comment (OlehHatsko) [reply EPlast/EPlast.WebApi/Controllers/GoverningBodiesController.cs:66]: I want to inform the client that it was deleted successfully. Is it better to return nothing?

File: EPlast/EPlast.WebApi/Controllers/GoverningBodiesController.cs (hunk 8 above)
Comment (OlehHatsko) [reply EPlast/EPlast.WebApi/Controllers/GoverningBodiesController.cs:103]: 200 means that the action was performed successfully. So, I think it`s ok. I haven`t found better solution

File: EPlast/EPlast.BLL/DTO/GoverningBody/GoverningBodyDocumentTypeDTO.cs [hunk 10]
```diff
@@ -0,0 +1,12 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Text;
+
+namespace EPlast.BLL.DTO.GoverningBody
+{
+    public class GoverningBodyDocumentTypeDTO
+    {
+        public int ID { get; set; }
```
Comment (atarason) [inline EPlast/EPlast.BLL/DTO/GoverningBody/GoverningBodyDocumentTypeDTO.cs:9]: Could you use Camelcase for naming? should be 'Id' not 'ID'

File: EPlast/EPlast.BLL/Services/GoverningBodies/GoverningBodiesService.cs [hunk 11]
```diff
@@ -78,11 +95,44 @@ private async Task UploadPhotoAsync(GoverningBodyDTO governingBody)
...
+            }
+
+            _repoWrapper.GoverningBody.Delete(governingBody);
+            await _repoWrapper.SaveAsync();
+            return governingBodyId;
+        }
+
+        public async Task<Dictionary<string, bool>> GetUserAccessAsync(string userId)
+        {
+            var userAcesses = await _securityModel.GetUserAccessAsync(userId);
```
Comment (atarason) [inline EPlast/EPlast.BLL/Services/GoverningBodies/GoverningBodiesService.cs:134]: Could you combine that two lines into one line?

File: EPlast/EPlast.BLL/Services/RegionBoard/RegionsBoardService.cs [hunk 12]
```diff
@@ -0,0 +1,26 @@
...
+
+        public RegionsBoardService(ISecurityModel securityModel)
+        {
+            _securityModel = securityModel;
+            _securityModel.SetSettingsFile(SecuritySettingsFile);
+        }
+
+        public async Task<Dictionary<string, bool>> GetUserAccessAsync(string userId)
+        {
+            var userAcesses = await _securityModel.GetUserAccessAsync(userId);
```
Comment (atarason) [inline EPlast/EPlast.BLL/Services/RegionBoard/RegionsBoardService.cs:22]: please, combine two lines into one line

File: EPlast/EPlast.DataAccess/Repositories/Interfaces/GoverningBody/IGoverningBodyAdministrationRepository.cs [hunk 13]
```diff
@@ -0,0 +1,11 @@
+﻿using System;
```
Comment (atarason) [inline EPlast/EPlast.DataAccess/Repositories/Interfaces/GoverningBody/IGoverningBodyAdministrationRepository.cs:1]: please, sort and remove usings

File: EPlast/EPlast.DataAccess/Repositories/Realizations/Base/RepositoryWrapper.cs [hunk 14]
```diff
@@ -10,6 +10,7 @@
 using EPlast.DataAccess.Repositories.Realizations.Region;
 using NLog.Extensions.Logging;
 using System.Threading.Tasks;
```
Comment (atarason) [inline EPlast/EPlast.DataAccess/Repositories/Realizations/Base/RepositoryWrapper.cs:12]: sort usings

File: EPlast/EPlast.Tests/Controllers/GoverningBodiesControllerTests.cs [hunk 15]
```diff
@@ -7,28 +7,35 @@
 using System.Collections.Generic;
 using System.Security.Principal;
```
Comment (atarason) [inline EPlast/EPlast.Tests/Controllers/GoverningBodiesControllerTests.cs:8]: sort usings

File: EPlast/EPlast.Tests/Controllers/RegionsBoardControllerTests.cs [hunk 16]
```diff
@@ -0,0 +1,41 @@
+﻿using System.Collections.Generic;
+using System.Threading.Tasks;
+using EPlast.BLL.Interfaces.RegionBoard;
```
Comment (atarason) [inline EPlast/EPlast.Tests/Controllers/RegionsBoardControllerTests.cs:3]: sort usings

File: EPlast/EPlast.WebApi/Controllers/GoverningBodiesController.cs (hunk 7 above)
Comment (atarason) [reply EPlast/EPlast.WebApi/Controllers/GoverningBodiesController.cs:65]: https://developer.mozilla.org/en-US/docs/Web/HTTP/Status

File: EPlast/EPlast.WebApi/Controllers/GoverningBodiesController.cs [hunk 17]
```diff
@@ -45,13 +50,63 @@ public async Task<IActionResult> Create(GoverningBodyDTO governingBodyDTO)
...
+        }
+
+        [HttpGet("LogoBase64/{logoName}")]
         [Authorize(AuthenticationSchemes = "Bearer")]
         public async Task<IActionResult> GetPhotoBase64(string logoName)
         {
-            var logoBase64 = await _governingBodiesService.GetLogoBase64(logoName);
+            if (logoName == null)
```
Comment (atarason) [inline EPlast/EPlast.WebApi/Controllers/GoverningBodiesController.cs:72]: Can we avoid using if else? I think we can have only if condition

**Review by atarason — CHANGES_REQUESTED** [review]

**Review by Oleh-Bodnar — APPROVED** [review]

**Review by veronika-kolosova — APPROVED** [review]

**Review by voroniak — APPROVED** [review]

**Review by atarason — APPROVED** [review]

**Pull request merged**